src/pathwayvis/services/mapoption.service.ts

```diff
@@ -376,6 +414,23 @@ export class MapOptionService {
     } else if (action.type === 'reaction:objective:undo') {
       shared.objectiveReaction = null;
     }
+    if (action.type === 'reaction:bounds:do') {
+      let index = shared.changedReactions.findIndex((reaction) => reaction.id === id);
+      if (index === -1) {
```

Comment:
Usually using `array[array.length]` is discouraged, but in this case I think it's much cleaner:

    shared.changedReactions[index > -1 ? index : shared.changedReactions.length] = {
      id,
      lower_bound: bounds.lower,
      upper_bound: bounds.uper,
    }
And please use the dot notation preferably